### ISSUE TITLE
assemble: Error with `remote_src=no`

### ISSUE THREAD

**Issue by dochang** [ansible/ansible-modules-core#3154]
##### Issue Type:
- Bug Report
##### Plugin Name:

assemble
##### Ansible Version:

```
ansible 2.0.1.0
  config file = /home/doc/.ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

```
[defaults]
retry_files_save_path = ~/.ansible/retry
roles_path = ~/.ansible/roles

[ssh_connection]
pipelining = True
```
##### Environment:

Local machine: Debian sid
Remote server: Debian jessie
##### Summary:

The assemble module with `remote_src=no` doesn't generate the target file.
##### Steps To Reproduce:

I've created a github [repo](https://github.com/dochang/ansible-assemble-test-1456221291).

Clone this repo, run the playbook with `smart` connection:

```
ansible-playbook -i your.remote.server, -c smart test.yml
```

Or with `local` connection:

```
ansible-playbook -i localhost, -c local test.yml
```
##### Expected Results:

`/tmp/assemble-test` is generated with the correct content.
##### Actual Results:

With _smart_ connection, assemble module fails with the message `Remote copy does not support recursive copy of directory: src`, and generates a file `~/src/tmpXXXXXX` on the remote server with the assembled content.

With _local_ connection, assemble module works if ansible needs to update the target file.  But if the target file has been already updated, ansible will put the assembled content into a file called `src` in the current directory on the local machine.

**Comment by diclophis** [comment]
I would also like to report this issue, it is currently blocking our upgrade to ansible-2.0

**Comment by bcoca** [comment]
this seems to be a consequence of adding `remote_src` to copy

**Comment by diclophis** [comment]
Thanks for the quick turn-around

**Comment by dochang** [comment]
I test this bug with Ansible 2.0.2.0 and the bug still occurs.

Please test & fix it again.

**Comment by urajio** [comment]
Confirm this bug with 2.0.2.0

**Comment by dochang** [comment]
@bcoca Please test this bug with 2.0.2.0. It still occurs.

**Comment by dochang** [comment]
This bug has been fixed in 2.1.0.0.  Confirmed.
